src/attribute_type.rs

```diff
@@ -396,567 +307,593 @@ impl SdpAttributeSsrc {
 }
 
 #[derive(Clone)]
-pub enum SdpAttributeValue {
-    Str(String),
-    Int(u64),
-    Vector(Vec<String>),
+pub enum SdpAttribute {
+    BundleOnly,
     Candidate(SdpAttributeCandidate),
+    EndOfCandidates,
     Extmap(SdpAttributeExtmap),
     Fingerprint(SdpAttributeFingerprint),
     Fmtp(SdpAttributeFmtp),
     Group(SdpAttributeGroup),
+    IceLite,
+    IceMismatch,
+    IceOptions(Vec<String>),
+    IcePwd(String),
+    IceUfrag(String),
+    Identity(String),
+    ImageAttr(String),
+    Inactive,
+    Label(String),
+    MaxMessageSize(u64),
+    MaxPtime(u64),
+    Mid(String),
     Msid(SdpAttributeMsid),
+    MsidSemantic(String),
+    Ptime(u64),
+    Rid(String),
+    Recvonly,
     RemoteCandidate(SdpAttributeRemoteCandidate),
     Rtpmap(SdpAttributeRtpmap),
     Rtcp(SdpAttributeRtcp),
     Rtcpfb(SdpAttributeRtcpFb),
+    RtcpMux,
+    RtcpRsize,
     Sctpmap(SdpAttributeSctpmap),
+    SctpPort(u64),
+    Sendonly,
+    Sendrecv,
     Setup(SdpAttributeSetup),
     Simulcast(SdpAttributeSimulcast),
     Ssrc(SdpAttributeSsrc),
+    SsrcGroup(String),
 }
 
-#[derive(Clone)]
-pub struct SdpAttribute {
-    pub name: SdpAttributeType,
-    pub value: Option<SdpAttributeValue>,
+impl FromStr for SdpAttribute {
+    type Err = SdpParserError;
+
+    fn from_str(line: &str) -> Result<SdpAttribute, SdpParserError> {
+        let tokens: Vec<_> = line.splitn(2, ':').collect();
+        let name = tokens[0].to_lowercase();
+        let val = match tokens.get(1) {
+            Some(x) => x.trim(),
+            None => ""
+        };
+        if tokens.len() > 1 {
+            match name.as_str() {
+                "bundle-only" | "end-of-candidates" | "ice-lite"
+                    | "ice-mismatch" |  "inactive" |  "recvonly"
+                    | "rtcp-mux" | "rtcp-rsize" | "sendonly"
+                    | "sendrecv" => {
+                        return Err(SdpParserError::Line{
+                            message: "This attribute is not allowed to have a value".to_string(),
+                            line: line.to_string()});
+                    },
+                _ => ()
+            }
+        }
```

Comment:
I think before matching the `name` we should check here the `!val.is_empty()`. That would simplify `string_or_empty()` and also ensure that all other parsers are guaranteed to have a value.